packages/react-core/src/components/Menu/MenuItem.tsx

```diff
@@ -102,7 +102,10 @@ export const MenuItem: React.FunctionComponent<MenuItemProps> = ({
     setFlyoutRef,
     disableHover
   } = React.useContext(MenuContext);
-  const Component = component || to ? 'a' : 'button';
+  let Component = !component && to ? 'a' : 'button';
+  if (component) {
+    Component = component as any;
+  }
```

Comment:
You can write `const Component = (!component && to ? 'a' : 'button') as any;`

Reply:
Will that preserve a defined `component` though? The first statement will fall through to setting `button` as the component. But if a `component` is given it should override that assignment.

Reply:
Oh sorry I missed that logic, you could do `const Component = (component || to ? 'a' : 'button') as any;` in that case.

Reply:
I also found this a little confusing.  Also why don't we just default component to `button` on line 80?

Reply:
I was trying to preserve the logic that let `to` set `component` to `a`. 

`component || to ? 'a' : 'button'` wouldn't keep a given `component` either since it's a if-else.

I could use `(to ? 'a' : component) as any` and default `component` to `button`.